src/graph/fixtures.ts

```diff
@@ -3,96 +3,152 @@ import {
   SavedItemStatus,
   Videoness,
 } from '../generated/graphql/types';
-import { ListItemObject, RestResponse } from './types';
+import { GraphSavedItemEdge, ListItemObject, RestResponse } from './types';
+export const seedDataRest: MockListItemObject = Object.freeze({
+  ids: ['1', '2'],
+  given_url: 'https://test.com',
+  given_title: 'title',
+  favorite: '1',
+  status: '0',
+  time_added: '1677818995',
+  time_updated: '1677818995',
+  time_read: '1677818995',
+  time_favorited: '1677818995',
+  resolved_title: 'title',
+  resolved_url: 'https://test.com',
+  title: 'title',
+  excerpt: 'excerpt',
+  is_article: '1',
+  is_index: '1',
+  has_video: '1',
+  has_image: '1',
+  word_count: '100',
+  lang: 'en',
+  time_to_read: 10,
+  amp_url: 'https://test.com',
+  top_image_url: 'https://test.com/image.jpg',
+});
 
-export const testSavedItemFragment = (id) => {
-  return {
-    id: id,
-    status: SavedItemStatus.Unread,
-    url: 'test.com',
-    isFavorite: true,
-    isArchived: true,
-    _createdAt: 1677818995,
-    _updatedAt: 1677818995,
-    favoritedAt: 1677818995,
-    archivedAt: 1677818995,
-    //can inject tagModel as parameter when mocking
-    tags: [
-      {
-        id: `tag-${id}-1`,
-        name: 'tag-name-1',
-      },
-      {
-        id: `tag-${id}-2`,
-        name: 'tag-name-2',
-      },
-    ],
-  };
+export type SavedItemFragment = Pick<
+  GraphSavedItemEdge['node'],
+  | '__typename'
+  | 'id'
+  | 'status'
+  | 'url'
+  | 'isFavorite'
+  | 'isArchived'
+  | '_createdAt'
+  | '_updatedAt'
+  | 'favoritedAt'
+  | 'archivedAt'
+>;
+export const mockSavedItemFragment: SavedItemFragment = {
+  __typename: 'SavedItem',
+  id: '1',
+  status: SavedItemStatus.Unread,
+  url: seedDataRest.given_url,
+  isFavorite: seedDataRest.favorite === '1',
+  isArchived: true, //todo: map status to archived
+  _createdAt: parseInt(seedDataRest.time_added),
+  _updatedAt: parseInt(seedDataRest.time_updated),
+  favoritedAt: parseInt(seedDataRest.time_favorited),
+  archivedAt: parseInt(seedDataRest.time_read),
+};
+export type ItemFragment = GraphSavedItemEdge['node']['item'];
```

Comment:
I'm assuming this translates to something like this in object notation `GraphSavedItemEdge.node.item`?

As in `item` here is a sub type on `GraphSavedItemEdge['node']` type?

Reply:
yes, `item` is a sub field in `GraphSavedItemEdge['node']` type
i attempted earlier to use the format you mentioned and it threw an error, probably coz generated types are like that too: https://github.com/Pocket/v3-proxy-api/blob/91651758b5430268cc497813b5a1ccb5aba9f718/src/graph/types.ts#L24